doc/api/esm.md

```diff
@@ -1130,18 +1130,45 @@ node --experimental-top-level-await b.mjs # works
 <!-- type=misc -->
 
 To customize the default module resolution, loader hooks can optionally be
-provided via a `--experimental-loader ./loader-name.mjs` argument to Node.js.
-
+provided via `--experimental-loader ./loader-name.mjs` arguments to Node.js
 When hooks are used they only apply to ES module loading and not to any
 CommonJS modules loaded.
 
+Loaders are modules which export functions and data which can be used by
+Node.js to modify the behavior of loading ES Modules. Loaders may be "chained"
+together by passing multiple `--experimental-loader` arguments to Node.js. A
+loader module will be imported once as a singleton.
+
+A hook has access to an additional `parentLoader` argument, which contains
+functions which call the hooks of the parent loader. When calling a parent hook,
+`parentLoader` **does not** need to be passed
+
+### <code>parentLoader</code>
+
+* {Object}
+  * `resolve` {Function}
+  * `getFormat` {Function}
+  * `getSource` {Function}
+  * `transformSource` {Function}
+
+When calling a parent hook, `parentLoader` **does not** need to be passed.
+
 ### Hooks
 
-#### <code>resolve</code> hook
+#### <code>resolve(specifier, parentURL, context, parentLoader)</code>
 
 > Note: The loaders API is being redesigned. This hook may disappear or its
 > signature may change. Do not rely on the API described below.
 
+* `specifier` {string}
+* `parentURL` {string}
+* `context` {Object}
+  * `conditions` {string[]}
+* `parentLoader` {Object}
+* Returns: {Object} If `null` is returned, Node.js will defer to the parent
+  loader.
+  * `url` {string}
```

Comment:
I think this is more readable in the code block where the previous version was.

Reply:
this aligns with our standard way of documenting APIs, which is also used for generating json metadata.